src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -9905,7 +9908,7 @@ private Task TDSExecuteRPCAddParameter(TdsParserStateObject stateObj, SqlParamet
                     stateObj.WriteByte(0);       // No schema
                 }
             }
-            else if (mt.IsCharType)
```

Comment:
This is needed for data writes, isnt it ?

Reply:
Yes. I included this change as it was needed as a fix for reads to work on top of write changes.